sprs/src/io.rs

```diff
@@ -160,10 +163,16 @@ where
         DataType::Integer => NumKind::Integer,
         DataType::Real => NumKind::Float,
         DataType::Complex => NumKind::Complex,
+        DataType::Pattern => NumKind::Pattern,
     };
-    if N::num_kind() != data_num_kind {
+    // any type can be convert to pattern
```

Comment:
Do you intend for float matrices to be interpreted as a pattern matrix? I would prefer if all such conversions were done explicitly by the user by e.g. mapping after reading.

Reply:
My consideration is that sometimes we have a lot of matrices in a folder, and we only care about the indices. Those matrices all have different types of data. It's hard for a user to read them all up and then convert them to the pattern.

Reply:
because the user can't know the real type before an error is reported.

Reply:
I have an idea. We can add a new function like `read_...._ignore_data(...)`, the old read function behavior like before(will return an error if the matrix is not originally Pattern), and the new `read_...._ignore_data(...)` always return CSMat<Pattern>, does this sound better?